### PR TITLE
Fixes and improves TGS revision info

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Fixes and improves the revision info when running the server on TGS. Currently, if you try run the server on TGS, you get this
image
Thats bad.

Now you get this:
image

Joining players will also be notified of testmerged PRs below the MOTD:
image

## Why It's Good For The Game
Git info should work if the server is on a different management platform

## Changelog
:cl: AffectedArc07
fix: Server now correctly reports revision info if running on the TGS toolkit. 
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
WIP:
- It **does** pull the right git hash now using DMAPI parameters
- It **cannot** pull the git hash date due to the `Repository` dir being the only one with a `.git` folder, and the game folders themselves not having a `.git` folder. I will likely end up modifying TGS itself with a new DMAPI release so that commit date can be put in.

File: code/datums/revision.dm [hunk 1]
```diff
@@ -11,30 +11,77 @@ GLOBAL_PROTECT(revision_info) // Dont mess with this
...
+  * * header - Should a header be sent too
+  */
+/datum/code_revision/proc/get_testmerge_chatmessage(header = FALSE)
+	var/list/msg = list()
+	if(header)
+		msg += "<span class='notice'>The following PRs are currently testmerged:</span>"
+
+	for(var/pr in GLOB.revision_info.testmerges)
+		var/datum/tgs_revision_information/test_merge/tm = pr
+		msg += "- PR <a href='[config.githuburl]/pull/[tm.number]'>#[tm.number] - [tm.title]</a>"
```
Comment (Cyberboss) [inline code/datums/revision.dm:81]:
```suggestion
		msg += "- PR <a href='[tm.url]'>#[tm.number] - [tm.title]</a>"
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/revision.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/datums/revision.dm:81]: Will do when I get around to sorting this again

**Comment by AffectedArc07** [comment]
image
Awaiting TGS release `v4.8.0` / DMAPI `v5.3.0`

File: code/datums/revision.dm [hunk 2]
```diff
@@ -52,9 +52,9 @@ GLOBAL_PROTECT(revision_info) // Dont mess with this
 			logmsgs += "The following PRs are testmerged:"
 			for(var/pr in testmerges)
 				var/datum/tgs_revision_information/test_merge/tm = pr
-				logmsgs += "PR #[tm.number] at commit [tm.pull_request_commit]"
+				logmsgs += "PR #[tm.number] at commit [tm.commit]"
```
Comment (Cyberboss) [inline code/datums/revision.dm:55]:
```suggestion
				logmsgs += "PR #[tm.number] at commit [tm.head_commit]"
```

File: code/datums/revision.dm [hunk 3]
```diff
@@ -52,9 +52,9 @@ GLOBAL_PROTECT(revision_info) // Dont mess with this
...
 				// Log these in blackbox so they can be attributed to round IDs easier in the future
-				SSblackbox.record_feedback("associative", "testmerged_prs", 1, list("number" = "[tm.number]", "commit" = "[tm.pull_request_commit]", "title" = "[tm.title]", "author" = "[tm.author]"))
+				SSblackbox.record_feedback("associative", "testmerged_prs", 1, list("number" = "[tm.number]", "commit" = "[tm.commit]", "title" = "[tm.title]", "author" = "[tm.author]"))
```
Comment (Cyberboss) [inline code/datums/revision.dm:57]:
```suggestion
				SSblackbox.record_feedback("associative", "testmerged_prs", 1, list("number" = "[tm.number]", "commit" = "[tm.head_commit]", "title" = "[tm.title]", "author" = "[tm.author]"))
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**